course/include/isometry.h

```diff
@@ -141,7 +143,57 @@ class Matrix3{
     private:
         // Ordered by rows.
         Vector3 r1_, r2_, r3_;
-};
+
+};  // Matrix3
+
+class Isometry {
+
+    public:
+        // Constructors
+        Isometry(const Vector3& translation_vector = Vector3(), 
+                 const Matrix3& rotation_matrix = Matrix3::kIdentity) :
+                translation_vector_{translation_vector},
+                rotation_matrix_{rotation_matrix} {}
+        Isometry(const Isometry& t) : 
+            translation_vector_{t.translation_vector_},
+            rotation_matrix_{t.rotation_matrix_} {}
+
+        // Static operations
+        static Isometry FromTranslation(const Vector3&);
+        static Isometry RotateAround(const Vector3&, const double);
+        static Isometry FromEulerAngles(const double, const double, const double);
+
+        // Operators
+        Matrix3& operator = (const Matrix3&);
+        Vector3 operator * (const Vector3&) const;
+        Isometry operator * (const Isometry&) const;
+        bool operator == (const Isometry&) const;
+
+        // Misc
+        Vector3 transform(const Vector3&) const;
+        Isometry compose(const Isometry&) const;
+
+        // Getters
+        Vector3 translation() const;
+        Matrix3 rotation() const;
+        Isometry inverse() const;
+        const double get_x_translation() const;
+        const double get_y_translation() const;
```

Comment:
What are these for? They don't seem necessary on the public interface.

Reply:
You are right. They were introduced because before the current implementation I had a different one. This is not necessary anymore.